### DISCUSSION TITLE
Custom field availability only for one or more specific name of an objet

### DISCUSSION THREAD

**Discussion by Djo91** [netbox-community/netbox#11295] [Q&A]
Hello,

Is it possible to create a custom field (and/or custom field group) that will be available only for one or more specific name of an objet ?

For example : 
- Create an objet "DCIM > Device Role" with the named "OLT"
- Create an objet "DCIM > Device Role" with the named "BNG"
- Create a custom field named "SLID" available for Model(s) "DCIM > Device Role"
- Make this custom field named "SLID" available for "OLT" role but not available for "BNG" role

Regards,

**Comment by dreng** [comment]
Short answer: No. And it wouldn't be a good idea from a database point of view either.

**Comment by mtinberg** [comment]
Longer answer would be to explore custom validators to enforce local rules or write custom Reports to find discrepancies and rely on automated audit notifications and ops handling to enforce local process.


—

Mark Tinberg ***@***.***>

Division of Information Technology-Network Services

University of Wisconsin-Madison


________________________________

**Comment by candlerb** [comment]
I think OP might be looking for different data entry forms depending on the role selected. Netbox does not support this. (The feature would be similar to Salesforce [Record Types](https://help.salesforce.com/s/articleView?id=sf.customize_recordtype.htm&type=5)).

**Comment by mtinberg** [comment]
You could do something like have a different Script for each role with your custom validation of custom fields, if the operational burden of building/maintaining a custom process is worth it.  I think in a lot of cases instead of spending more effort to prevent errors on the front-end, by having rigid, guided workflows for each data entry task, having a simpler process to catch errors with audit and allowing the tech to be open and flexible makes it more usable for the engineers and less likely they have to work around the process when some minor variation happens.  Where that line gets drawn between process flexibility and process correctness is different for different orgs though.


—

Mark Tinberg ***@***.***>

Division of Information Technology-Network Services

University of Wisconsin-Madison


________________________________
